includes/wc-functions.php

```diff
@@ -1011,7 +1011,8 @@
     if ( false === $is_purchasable || dokan_is_product_author( $product->get_id() ) ) {
         $is_purchasable = false;
     }
-    return $is_purchasable;
+
+    return apply_filters( 'dokan_vendor_own_product_purchase_restriction', $is_purchasable, $product );
```

Comment:
Adding doc block would be a nice addition for hooks.
Like: 
A short description about what the hook does
@since
@param